### PR TITLE
Adding Poisson Loss 

### PR DESCRIPTION
**Pull request by atreyamaj**
## Description
Adding the poisson regression loss function. 
Fix #(issue)
Refering Issue #2478 
This PR is intended for after the [Huber loss PR](https://github.com/deepchem/deepchem/pull/2479) is merged.

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [x] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.22.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

@rbharath  @alat-rights

### CODE REVIEW

File: deepchem/models/losses.py [hunk 1]
```diff
@@ -77,6 +94,28 @@ def loss(output, labels):
     return loss
 
 
+class PoissonLoss(Loss):
+  """The Poisson loss function is defined as the mean of the elements of y_pred - (y_true * log(y_pred) for an input of (y_true, y_pred).
+  Poisson loss is generally used for regression tasks where the data follows the poisson
+
```
Comment (alat-rights) [inline deepchem/models/losses.py:100]: Is this extra line intentional?

File: deepchem/models/losses.py [hunk 2]
```diff
@@ -77,6 +94,28 @@ def loss(output, labels):
...
+  def _compute_tf_loss(self, output, labels):
+    import tensorflow as tf
+    output, labels = _make_tf_shapes_consistent(output, labels)
+    loss = tf.keras.losses.Poisson(reduction='auto')
+    return loss(labels, output)
+
+  def _create_pytorch_loss(self):
+    import torch
+
+    def loss(output, labels):
```
Comment (alat-rights) [inline deepchem/models/losses.py:112]: Is this function supposed to be nested?

File: deepchem/models/tests/test_gcn.py [hunk 3]
```diff
@@ -22,17 +22,16 @@
 def test_gcn_regression():
```
Comment (alat-rights) [inline deepchem/models/tests/test_gcn.py:22]: Is this within the scope of this PR?

**Review by alat-rights — COMMENTED** [review]
Good work! Here are some comments

File: deepchem/models/losses.py (hunk 1 above)
Comment (atreyamaj) [reply deepchem/models/losses.py:100]: Good catch! I'll remove it ASAP.

File: deepchem/models/losses.py (hunk 2 above)
Comment (atreyamaj) [reply deepchem/models/losses.py:112]: Yes! I am following the conventions used in the other class definitions.

File: deepchem/models/tests/test_gcn.py (hunk 3 above)
Comment (atreyamaj) [reply deepchem/models/tests/test_gcn.py:22]: That change was made by flake8 I believe. This commit was part of the Huber loss PR from where I made a new branch.

Comment (alat-rights) [reply deepchem/models/tests/test_gcn.py:22]: Should be fine

File: deepchem/models/losses.py (hunk 2 above)
Comment (alat-rights) [reply deepchem/models/losses.py:111]: Sounds good, just checking.

**Comment by alat-rights** [comment]
LGTM! 😄

**Review by rbharath — APPROVED** [review]
LGTM! 

Will leave open for a day or two in case anyone has comments and then merge in

**Comment by atreyamaj** [comment]
@rbharath The changes in this PR have been merged into `main` through #2488 . Should I close this?

**Comment by rbharath** [comment]
Will go ahead and close it :)
